### PR TITLE
aaron/APPEALS-28367

### PR DESCRIPTION
**Pull request by Aaron-Willis**
Resolves [APPEALS-28367](https://jira.devops.va.gov/browse/APPEALS-28367)

# Description
As a developer,  need to add failed job alerts to ping in slack.

## Acceptance Criteria
Add Slack Alert to failing jobs for:
- [x] PopulateEndProductSyncQueueJob
- [x] PriorityEpSyncBatchProcessJob
- [x] BatchProcessRescueJob
- [x] Add attr_accessor for instance variables in batch_processes.rb

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### CODE REVIEW

File: app/jobs/batch_processes/batch_process_rescue_job.rb [hunk 1]
```diff
@@ -6,21 +6,17 @@
...
-  end
-
   def perform
     batches = BatchProcess.needs_reprocessing
     if batches.any?
       batches.each do |batch|
         begin
           batch.process_batch!
         rescue StandardError => error
-          Rails.logger.error("Error: #{error.inspect}, Job ID: #{JOB_ATTR&.job_id}, Job Time: #{Time.zone.now}")
```
Comment (nkirby-va) [inline app/jobs/batch_processes/batch_process_rescue_job.rb:20]: is `job_id` accessible in these jobs? is that defined somewhere in AJ?

File: spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb [hunk 2]
```diff
@@ -127,29 +155,63 @@
...
+          subject
+        end
+      end
+
+      it "a message that says 'Cannot Find Any Records to Batch' will be logged" do
+        expect(Rails.logger).to have_received(:info).with(
+          "PopulateEndProductSyncQueueJob is not able to find any batchable EPE records."\
+          "  Active Job ID: #{@job.job_id}.  Time: #{Time.zone.now}"
+        )
+      end
```
Comment (nkirby-va) [inline spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb:214]: should we have an 'expect slack to not receive notifications' on these sections in these PRs specs?

File: app/jobs/batch_processes/batch_process_rescue_job.rb (hunk 1 above)
Comment (Aaron-Willis) [reply app/jobs/batch_processes/batch_process_rescue_job.rb:20]: Yes, it is part of Active Job.

File: spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb (hunk 2 above)
Comment (Aaron-Willis) [reply spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb:216]: I added additional Slack Alert negative testing to RSPEC.

Comment (nkirby-va) [reply spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb:216]: should we add additional [here](https://github.com/department-of-veterans-affairs/caseflow/blob/cddef970b5235d8ee100b7e5d2bd753103bdaf7c/spec/jobs/batch_processes/priority_ep_sync_batch_process_job_spec.rb#L239) and [here](https://github.com/department-of-veterans-affairs/caseflow/blob/cddef970b5235d8ee100b7e5d2bd753103bdaf7c/spec/jobs/batch_processes/batch_process_rescue_job_spec.rb#L267)?

Comment (Aaron-Willis) [reply spec/jobs/priority_queues/populate_end_product_sync_queue_job_spec.rb:216]: batch_process_rescue_job_spec.rb & populate_end_product_sync_queue_job_spec.rb have been updated with the additional slack alert negative tests.

**Review by nkirby-va — DISMISSED** [review]

**Review by nkirby-va — APPROVED** [review]

**Pull request merged**